sandbox/pages/StatusWizardStepperPage.qml

```diff
@@ -0,0 +1,82 @@
+import QtQuick 2.14
+
+import StatusQ.Core 0.1
+import StatusQ.Components 0.1
+
+Item {
+    id: root
+    width: 800
+    height: 100
+
+    //Simulate animation between steps
+    property bool stepsCompleted: false
+    function reset() {
+        for (var i = 0; i < stepsListModel.count; i++) {
+            stepsListModel.setProperty(i, "loading", false);
+            stepsListModel.setProperty(i, "stepCompleted", false);
+        }
+    }
+
+    ListModel {
+        id: stepsListModel
+        ListElement {description:"Send Request"; loadingTime: 0; loading: false; stepCompleted: false}
+        ListElement {description:"Receive Response"; loadingTime: 2000; loading: false; stepCompleted: false}
+        ListElement {description:"Confirm Identity"; loadingTime: 2000; loading: false; stepCompleted: false}
+    }
+
+    Shortcut {
+        sequence: "Ctrl+R"
+        onActivated: {
+            if (!root.stepsCompleted) {
+                animate.running = true;
+            } else {
+                root.reset();
+            }
+        }
+    }
+
+    StatusBaseText {
+        id: indicLabel
+        anchors.horizontalCenter: parent.horizontalCenter
+        text: "Press Ctrl+R to run the animation"
+        font.pixelSize: 17
+    }
+
+    SequentialAnimation {
+        id: animate
+        ScriptAction {
+            script: {
+                stepsListModel.setProperty(0, "stepCompleted", true);
+                stepsListModel.setProperty(1, "loading", true);
+            }
+        }
+        PauseAnimation {
+            duration: 2100
+        }
+        ScriptAction {
+            script: {
+                stepsListModel.setProperty(1, "stepCompleted", true);
+                stepsListModel.setProperty(2, "loading", true);
+            }
+        }
+        PauseAnimation {
+            duration: 2100
+        }
+        ScriptAction {
+            script: {
+                stepsListModel.setProperty(2, "stepCompleted", true);
+                root.stepsCompleted = true;
+            }
+        }
+    }
+    //simulation code
+
+    StatusWizardStepper {
+        id: wizardStepper
+        width: (parent.width - 50)
+        anchors.top: indicLabel.bottom
+        anchors.topMargin: 16
+        anchors.horizontalCenter: parent.horizontalCenter
+        stepsModel: stepsListModel
```

Comment:
As mentioned above, I think it'd make sense for it to have an API to know at which step the stepper is. Maybe we can then also have dedicated components to `.complete(i)` and `uncomplete(i)` 

Wdyt?

Reply:
I don't know as I don't know how that should really work. We agreed with John to make it a fix 3 step for now and change it later if needed. Other than this the completeness could/should come from the backend which is the one to really validate the steps, not the component. This is why I used a ListModel for the simulation, that said, this is only my assumption as there is not much specified or any real use case yet. This should also respond to your other questions regarding `stepCompleted: true`, I'd expect all these should come from the backend.

Reply:
Thanks for your insights @alexandraB99 !
That makes sense for now. 
Regardless of what the backend does, I think it's also okay to propose an API for you **you** think this component **should** work and then take it from there.

I personally like to refer to Angular Material components that I've worked on in the past. They usually have nice thought-through APIs (although, it's very well possible that not everything works as nicely to QML).

If you need any inspiration for designing the API together with the backend, here the stepper component from Angular Material, maybe this is useful: https://material.angular.io/components/stepper/overview

Reply:
so you mean to invent or adapt an API without really knowing what the component should be doing exactly in order to just have something in place? what if we add stuff and we end up using only some percent of all these at the end? I am fine to add something, however I prefer to have the real use case to take reference from as I believe we're creating a component specific to status and not a generic framework that could be used anyrywhere like Angular.

Reply:
> so you mean to invent or adapt an API without really knowing what the component should be doing exactly in order to just have something in place?

No, I was making a **suggestion** and provide some reference material in case you want look into this.

We're not building anything here "just to have something in place" , which is why we have this discussion in the first place.

What I was trying to say is, that it's fine that you don't know what some backend is doing, because these StatusQ components should come with a well defined API regardless. So when authoring a new component based on designs, you have the freedom to come up with an API that makes sense (which is a good thing!) (see: https://github.com/status-im/StatusQ/issues/499#issuecomment-993302089 as an example). 

If you're not sure what a component **should** be doing, then usually common sense is a good place to start and also looking at what other component libraries with similar components have done (which is why I've linked the Angular Material as a reference). And of course, you can reach out and get opinions (which at latest stage happens here in a PR anyway).

> however I prefer to have the real use case to take reference from as I believe we're creating a component specific to status and not a generic framework that could be used anyrywhere like Angular

The use case reference is in the figma design linked in the original issue as of now. If more designs of the same component are needed to know what the component should be doing (other than representing "steps" in some wizard-like process), **and** common sense isn't enough, then we'll have to wait with implementing the thing.  

re: we're not a generic framework: 
Not sure what your point is here. The stepper component (just like any other) should come with an API that makes sense and is easy to use. You don't have to be a "framework" for that.

Look, I'm just trying to help here and make constructive suggestions. I'm not asking you to do things that waste your time. If you're not happy with certain suggestions for solving a problem that's perfectly fine, as long as we can discuss these things and clarify **why** things shouldn't be that way. 

But if you say the "why" is a missing backend and additional designs for the component are missing to understand what it should be doing, then all I'm saying is: "okay, no problem. It's perfectly fine to use common sense and come up with your own proposal. Don't let this stop you."

If you don't want to, that's fine too. Not trying to force anything here. Again, just suggestions.

Reply:
oh yeah please don't get me wrong either! I was just trying to understand why should I be adding an API that I think makes sense to me when I should be really waiting for the specification. That said, for example, to me might make sense that I expose step circle and bar colors, but at the end it's not necessary. This way, I would be adding complexity (+lines of code) in the component and they wouldn't be used at the end. What I meant with we're not a generic framework is exactly this, for example if that was the case, I would be exposing these properties just in case are needed in some specific use case of a specific project that makes use of my component but now since only status desktop client will be using this component, I would probably choose to keep the API to the necessary level according to the specifications, which are non in place yet :/ . I hope that makes sense.